src/ig/IGExporter.ts

```diff
@@ -330,68 +305,36 @@ export class IGExporter {
-            title: 'Home',
-            generation
-          });
-          return;
+        if (existsSync(inputIndexXMLPagesPath) || existsSync(inputIndexXMLPageContentPath)) {
+          generation = 'html';
         }
+        // Add user-provided index file to IG definition
+        this.ig.definition.page.page.push({
+          nameUrl: 'index.html',
+          title: 'Home',
+          generation
+        });
+        return;
       }
 
-      const base = this.isIgPubContext ? '' : path.basename(this.igDataPath);
-      const filePathString = path.join(base, 'input');
+      // Generate index file if indexPageContent is provided in config
+      const filePathString = 'input';
       const warning = warningBlock('<!-- index.md {% comment %}', '{% endcomment %} -->', [
         `To change the contents of this file, edit the "indexPageContent" attribute in the tank ${this.configName} file`,
         `or provide your own index file in the ${filePathString}${path.sep}pagecontent or ${filePathString}${path.sep}pages folder.`,
         'See: https://build.fhir.org/ig/FHIR/ig-guidance/using-templates.html#root.input'
       ]);
       const outputPath = path.join(pageContentExportPath, 'index.md');
       outputFileSync(outputPath, `${warning}${this.config.indexPageContent}`);
-      this.updateOutputLog(outputPath, [this.configPath], 'generated');
       logger.info(`Generated index.md based on "indexPageContent" in ${this.configName}.`);
     } else if (existsSync(inputIndexMarkdownPageContentPath)) {
-      if (this.shouldCopyFiles) {
-        ensureDirSync(pageContentExportPath);
-        this.copyWithWarningText(
-          inputIndexMarkdownPageContentPath,
-          path.join(pageContentExportPath, 'index.md')
-        );
-        logger.info(`Copied ${path.join(pageContentExportPath, 'index.md')}`);
-      }
     } else if (existsSync(inputIndexXMLPageContentPath)) {
       generation = 'html';
-      if (this.shouldCopyFiles) {
-        ensureDirSync(pageContentExportPath);
-        this.copyWithWarningText(
-          inputIndexXMLPageContentPath,
-          path.join(pageContentExportPath, 'index.xml')
-        );
-        logger.info(`Copied ${path.join(pageContentExportPath, 'index.xml')}`);
-      }
     } else if (existsSync(inputIndexMarkdownPagesPath)) {
-      if (this.shouldCopyFiles) {
-        ensureDirSync(pagesExportPath);
-        this.copyWithWarningText(
-          inputIndexMarkdownPagesPath,
-          path.join(pagesExportPath, 'index.md')
-        );
-        logger.info(`Copied ${path.join(pagesExportPath, 'index.md')}`);
-      }
     } else if (existsSync(inputIndexXMLPagesPath)) {
       generation = 'html';
-      if (this.shouldCopyFiles) {
-        ensureDirSync(pagesExportPath);
-        this.copyWithWarningText(inputIndexXMLPagesPath, path.join(pagesExportPath, 'index.xml'));
-        logger.info(`Copied ${path.join(pagesExportPath, 'index.xml')}`);
-      }
     } else {
```

Comment:
I wanted to call this section out as well because it's a little bit odd. I kept the empty `else if` cases because this test fails when they are removed. The fixture in that test has both an `index.md` and an `index.xml` file. Previously, it would fall into

```
else if (existsSync(inputIndexMarkdownPageContentPath))
```

which could keep the `generation` property as `'markdown'`.

When I simplify this to

```
} else if (existsSync(inputIndexMarkdownPagesPath) || existsSync(inputIndexXMLPagesPath)) {
      generation = 'html';
}
```

then that case is reached in the test because an `index.xml` file is present, and the test fails because it's expecting the `generation` to be `'markdown'` still. Switching the generation seems to swap which the IG Publisher displays, so I wanted to keep it the same if possible.

Reply:
Updated here and in the similar case a few lines above: f8706fe